### ISSUE TITLE
Ignore attribute not working properly

### ISSUE THREAD

**Issue by the-black-wolf** [JoshClose/CsvHelper#1585]
**Describe the bug**
When a reference property is market with `[Ignore]` attribute, the CswWriter still drills into reference. 

**To Reproduce**
1. Create classes A and B, have A contain some properties and a property of type B with `Ignore` attribute, B also has some properties
2. Create a list of As, setting both As and Bs to new instances
2. Use CsvWriter to write list of As

```csharp
await using var tw = File.CreateText(ofd.FileName);
await using var csv = new CsvWriter(tw, CultureInfo.CurrentCulture);
csv.Configuration.IgnoreReferences = true;
csv.WriteRecords(data);
```

**Expected behavior**
CSV containing only non-ignored properties items from A. The CSV actually contains properties from B, regardless of setting `IgnoreReferences` to true.
EDIT: Actually sorry, the IgnoreReference = true ignores them. Still the IgnoreAttribute is not working.

**Additional context**
I believe this is a regression, as far as I remember from previous versions this was not the case, ignore was stopping the drill down.

**Comment by JoshClose** [comment]
What version of CsvHelper are you using?

**Comment by JoshClose** [comment]
This seems to work for me. Can you modify this example to show your issue?

```cs
void Main()
{
	var records = new List<A>
	{
		new A 
		{ 
			Id = 1, 
			Name = "A one",
			B = new B
			{
				Id = 1,
				Name = "B one"
			}
		},
		new A 
		{ 
			Id = 2, 
			Name = "A two" ,
			B = new B
			{
				Id = 2,
				Name = "B two"
			}
		},
	};
	
	using (var writer = new StringWriter())
	using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))
	{
		csv.WriteRecords(records);
		
		writer.ToString().Dump();
	}
}

public class A
{	
	public int Id { get; set; }
	public string Name { get; set; }
	[Ignore]
	public B B { get; set; }
}

public class B
{
	public int Id { get; set; }
	public string Name { get; set; }
}
```
Results:

![image](https://user-images.githubusercontent.com/114971/94455310-855b7980-0178-11eb-9323-0ea2612a59b6.png)

**Comment by the-black-wolf** [comment]
@JoshClose  15.0.5 is on the current build. Let me get back to you on this.

**Comment by the-black-wolf** [comment]
Ok, this is defintiely the weirdest bug of this year. Your sample works ok, when I put A/B sample in my code it also works. When I put my data classes in your sample code it also works. But my classes in my code, still does not work ok. I am completely confused as to why this is happening. I literally minimized the command code to this:

```csharp
        public ICommand ExportToCsvCommand => new DelegateCommand(() =>
        {
            using var tw = File.CreateText(@"c:\test\sample.csv");
            using var csv = new CsvWriter(tw, CultureInfo.CurrentCulture);
            csv.WriteRecords(Output);
        });
```

and the output is still being drilled down into ignored references.

**Comment by the-black-wolf** [comment]
@JoshClose Ok, got it. It is definitely a bug, and it fires up when there are cascading objects and not all are covered by `Ignore`. See this example:

```csharp
using System;
using System.Collections.ObjectModel;
using System.Globalization;
using System.IO;
using CsvHelper;
using CsvHelper.Configuration.Attributes;

namespace TestCSV
{
    internal class Program
    {
        private static void Main()
        {
            var records = new ObservableCollection<A>
            {
                new A()
                {
                    IdA = 1,
                    NameA = "A one",
                    B = new B()
                    {
                        IdB = 10,
                        NameB = "B one",
                        C = new C()
                        {
                            IdC = 100,
                            NameC = "C one"
                        }
                    }
                },
                new A()
                {
                    IdA = 2,
                    NameA = "A two",
                    B = new B()
                    {
                        IdB = 20,
                        NameB = "B two",
                        C = new C()
                        {
                            IdC = 200,
                            NameC = "C two"
                        }
                    }
                }
            };

            using var writer = new StringWriter();
            using var csv = new CsvWriter(writer, CultureInfo.CurrentCulture);
            csv.WriteRecords(records);

            Console.WriteLine(writer.ToString());
        }

        public sealed class A
        {
            public int IdA { get; set; }
            public string NameA { get; set; }
            [Ignore] public B B { get; set; }
        }

        public sealed class B
        {
            public int IdB { get; set; }
            public string NameB { get; set; }
            public C C { get; set; }
        }

        public sealed class C
        {
            public int IdC { get; set; }
            public string NameC { get; set; }
        }

    }
}
```

The output it produces is:
```csv
IdA,NameA,IdC,NameC
1,A one,100,C one
2,A two,200,C two
```

Even though B is ignored (and so nothing from it, including C and its properties) the values from C get published. No longer sure it was a regression, we might actually have been generating faulty CSVs from the start (I still have to confirm this with the consumer side)

**Comment by JoshClose** [comment]
Same thing happens when reading. This might be intended, though seems strange. I'll have to look into it.

**Comment by the-black-wolf** [comment]
It shouldn't be included, if intended the convention should be reexamined. Especially since you might not (and often do not) have control over implementations of included ignored references.

**Comment by JoshClose** [comment]
I agree.

**Comment by DougSteiert** [comment]
Is this being worked on? I ran into this problem recently with it writing undesired attributes to a file and this took me very long to debug and figure out why - definitely an unwanted outcome with regards to writing CSV.

**Comment by JoshClose** [comment]
I'll take a look at it today. I think it should be a quick fix.

**Comment by JoshClose** [comment]
Based on unit tests, this was expected, but I don't see anything that makes it seem like it was intended. I'll push out a bug fix for this.

**Comment by JoshClose** [comment]
This is fixed in version 15.0.10 on NuGet.
